meson/post_install.py

```diff
@@ -1,10 +1,17 @@
 #!/usr/bin/env python3
 
-import os
+from os import path, environ
 import subprocess
 
-schemadir = os.path.join(os.environ['MESON_INSTALL_PREFIX'], 'share', 'glib-2.0', 'schemas')
+prefix = environ.get('MESON_INSTALL_PREFIX', '/usr/local')
+schemadir = path.join(environ['MESON_INSTALL_PREFIX'], 'share', 'glib-2.0', 'schemas')
+datadir = path.join(prefix, 'share')
+desktop_database_dir = path.join(datadir, 'applications')
 
-if not os.environ.get('DESTDIR'):
-	print('Compiling gsettings schemas...')
-subprocess.call(['glib-compile-schemas', schemadir])
+if not environ.get('DESTDIR'):
+    print('Compiling gsettings schemas…')
+    subprocess.call(['glib-compile-schemas', schemadir])
+    print('Updating desktop database…')
+    subprocess.call(['update-desktop-database', '-q', desktop_database_dir])
+    print('Updating icon cache…')
+    subprocess.call(['gtk-update-icon-cache', '-qtf', path.join(datadir, 'icons', 'hicolor')])
```

Comment:
Where is this logic coming from? We're asking the desktop to update it's icon cache? Is that common practice? Does this make sense with it being a flatpak?

Reply:
This could be replaced by https://mesonbuild.com/Gnome-module.html#gnomepost_install but that requires meson 0.57.0 as a minimum.

This is very common practice when installing normally, to /usr or /usr/local -- although flatpak may internally handle running this when needed.

Reply:
Yes this is a common practice when building the app locally with meson. Even if your app is deployed as a Flatpak you should keep the meson install config.